matcher.go

```diff
@@ -1,43 +1,61 @@
 package gostatsd
 
 import (
+	"regexp"
 	"strings"
 )
 
 type StringMatch struct {
 	test        string
 	invertMatch bool
 	prefixMatch bool
+	isRegex     bool
+	regex       *regexp.Regexp
 }
 
 type StringMatchList []StringMatch
 
 func NewStringMatch(s string) StringMatch {
+
 	invert := false
 	if strings.HasPrefix(s, "!") {
 		invert = true
 		s = s[1:]
 	}
 
-	prefix := false
-	if strings.HasSuffix(s, "*") {
-		prefix = true
-		s = s[0 : len(s)-1]
+	regex := false
+	var compiledRegex *regexp.Regexp = nil
+	if strings.HasPrefix(s, "regex:") {
+		regex = true
+		s = s[6:]
+		compiledRegex, _ = regexp.Compile(s)
```

Comment:
Don't ignore errors - it would be better to propagate the error out, but that's a bit of a pain to plumb through the system. You can use `regexp.MustCompile`.  Since this is initialization only code, it's ok (but not ideal) to fail fast with a panic.